### ISSUE TITLE
Creating historical records for models with M2M fields to `"self"` causes `FieldError`

### ISSUE THREAD

**Issue by ddabble** [jazzband/django-simple-history#1126]
**Describe the bug**
*See title.*

**To Reproduce**
1. Given the following model:
   ```python
   class Person(models.Model):
       relations = models.ManyToManyField("self")
       history = HistoricalRecords(m2m_fields=[relations])
   ```
2. Run the following code (which should also create a historical record for the `Person` object):
   ```python
   Person.objects.create()
   ```
3. This will produce the following error:
   ```
   django.core.exceptions.FieldError: Cannot resolve keyword 'person' into field. Choices are: from_person, from_person_id, id, to_person, to_person_id
   ```

**Expected behavior**
That a model object and associated historical record were successfully created, and that the error was not raised.

**Environment (please complete the following information):**
 - OS: Windows 11 22H2
 - Django Simple History Version: [the current `master` branch](https://github.com/jazzband/django-simple-history/tree/636bcbc46d473862c000101ef040e4eda693117f)
 - Django Version: 4.1.6
 - Database Version: SQLite 3.38.4

**Comment by Vitalick** [comment]
I did this fix in my project.

Change
https://github.com/jazzband/django-simple-history/blob/321b78354f36ccea42375a5b698596412e10cf72/simple_history/models.py#L673

To

```python
rows = through_model.objects.all()
if field.related_model == original_instance.__class__:
    rows = rows.filter(models.Q(**{
        "from_"+through_field_name: instance,
    }) | models.Q(**{
        "to_"+through_field_name: instance,
    }))
else:
    rows = rows.filter(**{through_field_name: instance})
```

**Comment by nciri00** [comment]
With @Vitalick solution implemented I realized the change history is only logged in one direction. 

For example, using the model above, if a relationship between person1 and person2 is created via person1, a historical record is only created for person1. To generate to corresponding record for person2, it must be saved separately.

**Comment by nciri00** [comment]
On Read the Docs it says this issue would be fixed in the 3.4 release; however, I don't see the fixes for this issue in the release.

**Comment by ddabble** [comment]
@nciri00 Yeah, seems like that changelog entry was added to the wrong section by mistake - it's actually unreleased 😅 Commit https://github.com/jazzband/django-simple-history/pull/1262/commits/bf22f555a47cf7b59d955de4b9a101a3e85aa123 will fix that.

**Comment by ddabble** [comment]
This is now included in version 3.5.0 🙂
